src/System.Net.Http/tests/FunctionalTests/IdnaProtocolTests.cs

```diff
@@ -0,0 +1,118 @@
+        {
+            await LoopbackServer.CreateServerAsync(async (server, url) =>
+            {
+                using (HttpClient client = new HttpClient(CreateHttpClientHandler()))
+                {
+                    var request = new HttpRequestMessage(HttpMethod.Get, url);
+                    request.Headers.Host = s_IdnaHost;
+                    request.Headers.Referrer = s_IdnaUri;
+                    Task<HttpResponseMessage> getResponseTask = client.SendAsync(request);
+                    Task<List<string>> serverTask = LoopbackServer.ReadRequestAndSendResponseAsync(server, LoopbackServer.DefaultHttpResponse);
+
+                    await TestHelper.WhenAllCompletedOrAnyFailed(getResponseTask, serverTask);
+
+                    List<string> requestLines = await serverTask;
+
+                    Assert.Contains($"Host: {s_EncodedHostName}", requestLines);
+                    Assert.Contains($"Referer: {s_EncodedUri}", requestLines);
+                }
+            });
+        }
+
+        [ActiveIssue(26355)] // We aren't doing IDNA decoding properly
+        [Fact]
+        public async Task InternationalResponseHeaderValues_UsesIdnaDecoding_Success()
+        {
+            await LoopbackServer.CreateServerAsync(async (server, url) =>
+            {
+                HttpClientHandler handler = CreateHttpClientHandler();
+                handler.AllowAutoRedirect = false;
+
+                using (HttpClient client = new HttpClient(handler))
+                {
+                    Task<HttpResponseMessage> getResponseTask = client.GetAsync(url);
+                    Task<List<string>> serverTask = LoopbackServer.ReadRequestAndSendResponseAsync(server, 
+                        $"HTTP/1.1 302 Redirect\r\nLocation: {s_EncodedUri}\r\n\r\n");
+
+                    await TestHelper.WhenAllCompletedOrAnyFailed(getResponseTask, serverTask);
+
+                    HttpResponseMessage response = await getResponseTask;
+                    Console.WriteLine($"Location header={response.Headers.Location}");
+                    Console.WriteLine($"Location header.Host={response.Headers.Location.Host}");
+                    Console.WriteLine($"Location header.IdnHost={response.Headers.Location.IdnHost}");
+                }
+            });
+        }
+
+        sealed class SimpleWebProxy : IWebProxy
+        {
+            private Uri _proxyUri;
+
+            public SimpleWebProxy(Uri proxyUri)
+            {
+                _proxyUri = proxyUri;
+            }
+
+            public ICredentials Credentials { get => null; set => throw new NotImplementedException(); }
+            public Uri GetProxy(Uri destination) => _proxyUri;
+            public bool IsBypassed(Uri host) => false;
+        }
+    }
+
+    public sealed class DefaultHandler_IdnaProtocolTests : IdnaProtocolTests
+    {
+    }
+
+    public sealed class ManagedHandler_IdnaProtocolTests : IdnaProtocolTests
```

Comment:
This should go in the ManagedHandlerTests.cs file, all our managed handler tests are organized there.

Reply:
See my response to Stephen below